.github/draft_release_notes.sh

```diff
@@ -0,0 +1,54 @@
+#! /bin/sh
+
+# this script assumes the GITHUB_TOKEN and PREVIOUS_TAG environment variables have been set;
+# it produces a 'Changes.md' file as its final output;
+# the file 'last-300-prs-with-release-note.txt' that is produces is intermediate data; it is not
+# pruned for now to assist development of the release notes process (we are still curating all this)
+
+sudo apt-get -y install jq
+sudo apt-get -y install wget
+curl -L https://github.com/github/hub/releases/download/v2.14.2/hub-linux-amd64-2.14.2.tgz | tar xzf -
+PWD=`pwd`
+export PATH=$PWD/hub-linux-amd64-2.14.2/bin:$PATH
+git fetch --all --tags --prune --force
+echo -e "# Insert Title\n" > Changes.md
+echo -e "## Features\n\n## Fixes\n\n## Backwards incompatible changes\n\n## Docs\n\n## Misc\n\n## Thanks" >> Changes.md    - name: Draft Release
+# this effectively gets the commit associated with github.event.inputs.tags
+COMMON_ANCESTOR=$(git merge-base $PREVIOUS_TAG HEAD)
+# in theory the new tag has not been created yet; do we want another input that specifies the existing
+# commit desired for drafting the release? for now, we are using HEAD in the above git merge-base call
+# and PR cross referencing below
+
+# use of 'hub', which is an extension of the 'git' CLI, allows for pulling of PRs, though we can't search based on commits
+# associated with those PRs, so we grab a super big number, 300, which should guarantee grabbing all the PRs back to
+# github.events.inputs.tags; we use grep -v to filter out release-note-none and release-note-action-required
+hub pr list --state merged -L 300 -f "%sm;%au;%i;%t;%L%n" | egrep ", release-note|release-note," | grep -v release-note-none | grep -v release-note-action-required > last-300-prs-with-release-note.txt
```

Comment:
Again, `shellcheck` says that `egrep` is not standard. I never had an issue, but I just checked and on my system it is an alias. If `grep -E` does the trick too, we could consider to change it.

Reply:
sure I'll give that a go and change it if it works

Reply:
update it does work ... part of next push